### ISSUE TITLE
Adding Post Comments Form block to post causes validation error for `comment-reply` enqueued script

### ISSUE THREAD

**Issue by westonruter** [Automattic/amp-wp#6231]
## Bug Description

As part of Full Site Editing, Gutenberg has a “Post Comments Form” block:

> ![image](https://user-images.githubusercontent.com/134745/118200473-f69b2480-b409-11eb-8444-4e785380405c.png)

This renders the standard comments form inside the post content:

> ![image](https://user-images.githubusercontent.com/134745/118200488-0286e680-b40a-11eb-850a-ef577daeae1f.png)

When viewing such a post on the frontend, a validation error is reported:

> ![image](https://user-images.githubusercontent.com/134745/118200522-20ece200-b40a-11eb-8df9-4f0e66d0b54f.png)

Normally the `comment-reply` form is dequeued when the `singular` template is loaded:

https://github.com/ampproject/amp-wp/blob/1c7728ccbfc880118130306729a3f0f1d4768032/includes/class-amp-theme-support.php#L901-L906

This doesn't work for the block, however, because the script is [enqueued as part of the render callback](https://github.com/WordPress/gutenberg/blob/3a24d55dd76baad4bedba7b2dd23b9b39a57f072/packages/block-library/src/post-comments-form/index.php#L31-L32).

This is the second block we've seen in core to enqueue a script, the first being the File block which we added support for in #6112. We may need to take a similar approach where we dequeue the script just-in-time at `wp_print_scripts `/`wp_print_footer_scripts`. For example:

```diff
--- a/includes/class-amp-theme-support.php
+++ b/includes/class-amp-theme-support.php
@@ -898,12 +898,12 @@ static function() {
 			},
 			0
 		);
-		add_action(
-			'wp_enqueue_scripts',
-			static function() {
-				wp_dequeue_script( 'comment-reply' ); // Handled largely by AMP_Comments_Sanitizer and *reply* methods in this class.
-			}
-		);
+
+		$dequeue_comment_reply = static function() {
+			wp_dequeue_script( 'comment-reply' ); // Handled largely by AMP_Comments_Sanitizer and *reply* methods in this class.
+		};
+		add_action( 'wp_print_scripts', $dequeue_comment_reply, 0 );
+		add_action( 'wp_print_footer_scripts', $dequeue_comment_reply, 0 );
 	}
 
 	/**
```

_Note also that the source of this script is erroneously being reported as Twenty Twenty-One when it should be Core._

## Steps to reproduce

0. Activate Gutenberg.
1. Add the Post Comments Form to the content
2. Validate the AMP page.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by delawski** [comment]
**QA Passed**

✅ After activating the [TT1 Blocks](https://wordpress.org/themes/tt1-blocks/) theme and adding a Post Comments Form to a post, I got no "Invalid script" validation issue:

![Screenshot 2021-12-01 at 17 25 18](https://user-images.githubusercontent.com/478735/144273163-957c2000-2607-4baa-8471-87df6bf9e128.png)

Note that there was no Post Comments Form available in the regular Twenty Twenty-One theme. Following the [instruction](https://make.wordpress.org/test/handbook/full-site-editing-outreach-experiment/how-to-test-fse/) on make.wordpress.org, I went ahead and installed the TT1 Blocks theme.

_Tested on AMP Version 2.2.0-alpha-20211123T020732Z-5405daa_

**Comment by delawski** [comment]
@westonruter While QA'ing this issue, I've noticed a potential new issue.

The TT1 Blocks theme replaces the default theme header with a block-based template part. The template part uses a Navigation block which seems to be enqueuing yet another script, `gutenberg/build/block-library/blocks/navigation/view.min.js`. Similarly to `comment-reply.js`, it causes an AMP validation error:

![Screenshot 2021-12-01 at 17 35 53](https://user-images.githubusercontent.com/478735/144274704-cf8d52cb-91ac-4d23-9ca5-2708d1f078ca.png)

Should I create a new bug report for that or will we take care of FSE and TT1 Blocks theme later (as I think there might be more blocks causing similar errors)?

**Comment by westonruter** [comment]
@delawski I think this will be addressed by #6319 for #6649. So I think this is captured already and it's being tracked for v2.3.
